### PR TITLE
Added flag to pass in CAcerts

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Add CAcerts commandline flag to pass in the file containing the CA certificates which client passes to talk to admin

Testing done:
1.  Modified root_darwin.go to not load any systemRoot certs.
2. Downloaded root  cert from https://opta.gcp.union.ai/console
3. flytectl get projects --admin.caCerts ~/Downloads/opta-gcp-union-ai.pem
    which fetched the projects from admin
4.  Downloaded root  chain cert https://opta.gcp.union.ai/console
5. flytectl get projects --admin.caCerts ~/Downloads/opta-gcp-union-ai-chain.pem
    which fetched the projects from admin

With no systemRoot certs (Modifying root_darwin.go in go crypto https://go.dev/src/crypto/x509/root_darwin.go)

```
flytectl    get projects --logger.level 6
{"json":{"src":"viper.go:398"},"level":"debug","msg":"Config section [storage] updated. No update handler registered.","ts":"2021-12-22T12:05:40+05:30"}
{"json":{"src":"viper.go:398"},"level":"debug","msg":"Config section [root] updated. No update handler registered.","ts":"2021-12-22T12:05:40+05:30"}
{"json":{"src":"viper.go:400"},"level":"debug","msg":"Config section [admin] updated. Firing updated event.","ts":"2021-12-22T12:05:40+05:30"}
{"json":{"src":"client.go:180"},"level":"error","msg":"failed to initialize token source provider. Err: rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"","ts":"2021-12-22T12:05:51+05:30"}
{"json":{"src":"client.go:185"},"level":"warning","msg":"Starting an unauthenticated client because: can't create authenticated channel without a TokenSourceProvider","ts":"2021-12-22T12:05:51+05:30"}
{"json":{"src":"client.go:58"},"level":"info","msg":"Initialized Admin client","ts":"2021-12-22T12:05:51+05:30"}
Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate signed by unknown authority"
{"json":{"src":"main.go:13"},"level":"error","msg":"rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"","ts":"2021-12-22T12:05:51+05:30"}
Exiting.

```

By passing the root CA

```
flytectl    get projects --logger.level 6  --admin.caCerts ~/Downloads/opta-gcp-union-ai-chain.pem
{"json":{"src":"viper.go:398"},"level":"debug","msg":"Config section [storage] updated. No update handler registered.","ts":"2021-12-22T12:06:21+05:30"}
{"json":{"src":"viper.go:398"},"level":"debug","msg":"Config section [root] updated. No update handler registered.","ts":"2021-12-22T12:06:21+05:30"}
{"json":{"src":"viper.go:400"},"level":"debug","msg":"Config section [admin] updated. Firing updated event.","ts":"2021-12-22T12:06:21+05:30"}
{"json":{"src":"client.go:180"},"level":"error","msg":"failed to initialize token source provider. Err: rpc error: code = Unimplemented desc = unknown service flyteidl.service.AuthMetadataService","ts":"2021-12-22T12:06:22+05:30"}
{"json":{"src":"client.go:185"},"level":"warning","msg":"Starting an unauthenticated client because: can't create authenticated channel without a TokenSourceProvider","ts":"2021-12-22T12:06:22+05:30"}
{"json":{"src":"client.go:58"},"level":"info","msg":"Initialized Admin client","ts":"2021-12-22T12:06:22+05:30"}
{"json":{"src":"project.go:94"},"level":"debug","msg":"Retrieved 3 projects","ts":"2021-12-22T12:06:23+05:30"}
 --------------- --------------- --------------------------- 
| ID            | NAME          | DESCRIPTION               |
 --------------- --------------- --------------------------- 
| flyteexamples | flyteexamples | flyteexamples description |
 --------------- --------------- --------------------------- 
| flytetester   | flytetester   | flytetester description   |
 --------------- --------------- --------------------------- 
| flytesnacks   | flytesnacks   | flytesnacks description   |
 --------------- --------------- --------------------------- 
```

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/1888

## Follow-up issue
_NA_

### CODE REVIEW

File: clients/go/admin/config.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ type Config struct {
 	Endpoint              config.URL      `json:"endpoint" pflag:",For admin types, specify where the uri of the service is located."`
 	UseInsecureConnection bool            `json:"insecure" pflag:",Use insecure connection."`
 	InsecureSkipVerify    bool            `json:"insecureSkipVerify" pflag:",InsecureSkipVerify controls whether a client verifies the server's certificate chain and host name. Caution : shouldn't be use for production usecases'"`
+	CAcerts               string          `json:"caCerts" pflag:",Use specified certificate file to verify the admin server peer."`
```
Comment (wild-endeavor) [inline clients/go/admin/config.go:43]: can you make this singular?  just CAcert?

File: clients/go/admin/cert_loader.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"io/ioutil"
+
+	"crypto/x509"
+)
+
+// readCACerts from the passed in file at certLoc and return certpool.
+func readCACerts(certLoc string) (*x509.CertPool, error) {
+	rootPEM, err := ioutil.ReadFile(certLoc)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read from %v  file due to %v", certLoc, err)
```
Comment (wild-endeavor) [inline clients/go/admin/cert_loader.go:14]:
```suggestion
		return nil, fmt.Errorf("unable to read from %v file due to %v", certLoc, err)
```

File: clients/go/admin/config.go (hunk 1 above)
Comment (EngHabu) [inline clients/go/admin/config.go:43]:
```suggestion
	CACertFilePath               string          `json:"caCerts" pflag:",Use specified certificate file to verify the admin server peer."`
```

**Review by EngHabu — COMMENTED** [review]
Looks good! Let's get JPM to give a shot before merging? @wild-endeavor

File: clients/go/admin/config.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply clients/go/admin/config.go]: Made it CACertFilePath  as per this discussion https://github.com/flyteorg/flyteidl/pull/242#discussion_r773439741

Comment (pmahindrakar-oss) [reply clients/go/admin/config.go]: Also the flag i felts to keep caCerts since it can also contain a chain of certs to the rootCA  in the same file.
Let me know if you still think we need to make it singular. 
eg : opta-gcp-union-ai-chain.pem
        pta-gcp-union-ai.pem

File: clients/go/admin/config.go [hunk 3]
```diff
@@ -40,6 +40,7 @@ type Config struct {
 	Endpoint              config.URL      `json:"endpoint" pflag:",For admin types, specify where the uri of the service is located."`
 	UseInsecureConnection bool            `json:"insecure" pflag:",Use insecure connection."`
 	InsecureSkipVerify    bool            `json:"insecureSkipVerify" pflag:",InsecureSkipVerify controls whether a client verifies the server's certificate chain and host name. Caution : shouldn't be use for production usecases'"`
+	CACertFilePath        string          `json:"caCert" pflag:",Use specified certificate file to verify the admin server peer."`
```
Comment (katrogan) [inline clients/go/admin/config.go:43]: nit: should the json also match the variable name? e.g. `caCertFilePath` ?

**Review by katrogan — DISMISSED** [review]

File: clients/go/admin/config.go (hunk 3 above)
Comment (pmahindrakar-oss) [reply clients/go/admin/config.go]: Done. Had kept caCert to be in sync with flyte-cli

**Review by evalsocket — APPROVED** [review]

File: clients/go/admin/config.go (hunk 3 above)
Comment (katrogan) [reply clients/go/admin/config.go]: ah I see, sorry to be annoying!

**Review by katrogan — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]
